### PR TITLE
IJSDK-615: Fixed missing <depends> elements, refactored obsolete kotlin_demo.

### PR DESCRIPTION
**Pull request by JohnHake**

### CODE REVIEW

File: code_samples/facet_basics/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -30,6 +30,10 @@
   <!-- Compatible with the following versions of IntelliJ Platform -->
   <idea-version since-build="173"/>
 
+  <!-- Product and plugin compatibility requirements -->
+  <depends>com.intellij.modules.lang</depends>
+  <!-- Might use java -->
```
Comment (YannCebron) [inline code_samples/facet_basics/resources/META-INF/plugin.xml]: might use Java?!

File: code_samples/gradle_plugin_demo/build.gradle [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+}
+
+// See https://github.com/JetBrains/gradle-intellij-plugin/
+intellij {
+  // Define IntelliJ Platform API version to use for building this plugin
+  version '2019.1'
+}
+
+patchPluginXml {
+  untilBuild = '201.*'
```
Comment (YannCebron) [inline code_samples/gradle_plugin_demo/build.gradle]: typo?

File: code_samples/plugin_sample/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -16,8 +16,8 @@
 
   <!-- http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/plugin_compatibility.html -->
   <!-- The unique identifiers of the plugins on which this plugin depends -->
-  <!-- com.intellij.modules.lang is included in every IntelliJ-based IDE, plugin will be available as well -->
-  <depends>com.intellij.modules.lang</depends>
+  <!-- com.intellij.modules.platform is included in every IntelliJ-based IDE, plugin will be available as well -->
```
Comment (YannCebron) [inline code_samples/plugin_sample/resources/META-INF/plugin.xml]: please do not change this. "lang" is available in all products

Comment (JohnHake) [reply code_samples/plugin_sample/resources/META-INF/plugin.xml]: Substitute "lang" for "platform" everywhere "platform" is used.

**Pull request merged**